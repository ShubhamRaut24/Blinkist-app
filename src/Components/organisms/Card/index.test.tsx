import React from 'react';
import BookCard from ".";
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom'
const props = {
    id: 0,
    imgHeight: 300,
    url: '/assets/book.png',
    bookName: "Being Boss",
    writerName: "Kathleen Shannon and Emily...",
    timeRead: "13-minute read",
    width: 350,
    inLibrary: false,
};

test('BookCard', () => {
    const component = renderer.create(
        <BrowserRouter>
            <BookCard {...props}></BookCard>
        </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
});