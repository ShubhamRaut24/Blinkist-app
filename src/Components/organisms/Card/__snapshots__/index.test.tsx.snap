// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-v47ndi-MuiPaper-root"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "borderRadius": "8px",
      "margin": "16px 16px 16px",
    }
  }
>
  <a
    href="/book-info/undefined"
    onClick={[Function]}
  >
    <img
      alt="not found"
      className="MuiBox-root css-bohtpt"
      src="/assets/book.png"
      style={
        Object {
          "borderRadius": "8px 8px 0 0",
        }
      }
    />
  </a>
  <div
    className="MuiBox-root css-cxmp86"
    style={
      Object {
        "padding": "12px",
      }
    }
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1 css-1npw2q9-MuiTypography-root"
      style={
        Object {
          "fontSize": "18px",
          "fontWeight": "bold",
        }
      }
      variant1="subtitle1"
    >
      Being Boss
    </h6>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ov1z6c-MuiTypography-root"
      variant1="body1"
    >
      Kathleen Shannon and Emily...
    </p>
    <div
      className="MuiBox-root css-8owaep"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <div
        className="MuiBox-root css-old1by"
        data-testid="icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AccessTimeIcon"
          focusable="false"
          style={
            Object {
              "color": "#6D787E",
              "fontSize": "20px",
              "marginTop": "4px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
          <path
            d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
          />
        </svg>
      </div>
      <span
        className="MuiTypography-root MuiTypography-caption2 css-1fo51um-MuiTypography-root"
        variant1="caption2"
      >
        13-minute read
      </span>
    </div>
    <div
      className="MuiBox-root css-ckyhk3"
    >
      
    </div>
  </div>
  <div
    className="MuiBox-root css-79elbk"
  >
    <div
      className="MuiBox-root css-1ah5qgj"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": "0 0 8px 8px",
          "borderTop": "1px solid lightGrey",
          "display": "flex",
          "height": "48px",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <div
        className="MuiBox-root css-old1by"
        data-testid="icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body1 css-nb1x83-MuiTypography-root"
        variant1="body1"
      >
        Add to library
      </p>
    </div>
  </div>
</div>
`;
